### ISSUE TITLE
Error in calling dask.multiprocessing.get

### ISSUE THREAD

**Issue by antgr** [dask/dask#1172]
I installed dask,
dask                      0.9.0                    py27_2    conda-forge

and following the tutorial, I get error.

```python
In [5]: from dask.multiprocessing import get

In [6]: results = get(dsk, ['print1', 'print2'])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-a0eeda767e16> in <module>()
----> 1 results = get(dsk, ['print1', 'print2'])

/home/antgr/anaconda2/lib/python2.7/site-packages/dask/multiprocessing.pyc in get(dsk, keys, optimizations, num_workers, func_loads, func_dumps, **kwargs)
     76         # Run
     77         result = get_async(apply_async, len(pool._pool), dsk3, keys,
---> 78                            queue=queue, get_id=_process_get_id, **kwargs)
     79     finally:
     80         if cleanup:

/home/antgr/anaconda2/lib/python2.7/site-packages/dask/async.pyc in get_async(apply_async, num_workers, dsk, result, cache, queue, get_id, raise_on_exception, rerun_exceptions_locally, callbacks, **kwargs)
    492             f(key, res, dsk, state, worker_id)
    493         while state['ready'] and len(state['running']) < num_workers:
--> 494             fire_task()
    495 
    496     # Final reporting

/home/antgr/anaconda2/lib/python2.7/site-packages/dask/async.pyc in fire_task()
    463         # Submit
    464         apply_async(execute_task, args=[key, dsk[key], data, queue,
--> 465                                         get_id, raise_on_exception])
    466 
    467     # Seed initial tasks into the thread pool

/home/antgr/anaconda2/lib/python2.7/site-packages/toolz/functoolz.pyc in __call__(self, *args, **kwargs)
    217     def __call__(self, *args, **kwargs):
    218         try:
--> 219             return self._partial(*args, **kwargs)
    220         except TypeError:
    221             # If there was a genuine TypeError

/home/antgr/anaconda2/lib/python2.7/site-packages/dask/multiprocessing.pyc in pickle_apply_async(apply_async, func, args, kwds, func_loads, func_dumps)
     96     dumps = func_dumps or _globals.get('func_dumps') or _dumps
     97     sfunc = dumps(func)
---> 98     sargs = dumps(args)
     99     skwds = dumps(kwds)
    100     return apply_async(curry(apply_func, loads=func_loads),

/home/antgr/anaconda2/lib/python2.7/site-packages/dask/multiprocessing.pyc in _dumps(x)
     23 
     24 def _dumps(x):
---> 25     return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
     26 
     27 _loads = pickle.loads

/home/antgr/anaconda2/lib/python2.7/site-packages/cloudpickle/cloudpickle.pyc in dumps(obj, protocol)
    627 
    628     cp = CloudPickler(file,protocol)
--> 629     cp.dump(obj)
    630 
    631     return file.getvalue()

/home/antgr/anaconda2/lib/python2.7/site-packages/cloudpickle/cloudpickle.pyc in dump(self, obj)
    105         self.inject_addons()
    106         try:
--> 107             return Pickler.dump(self, obj)
    108         except RuntimeError as e:
    109             if 'recursion' in e.args[0]:

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in dump(self, obj)
    222         if self.proto >= 2:
    223             self.write(PROTO + chr(self.proto))
--> 224         self.save(obj)
    225         self.write(STOP)
    226 

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in save(self, obj)
    284         f = self.dispatch.get(t)
    285         if f:
--> 286             f(self, obj) # Call unbound method with explicit self
    287             return
    288 

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in save_list(self, obj)
    604 
    605         self.memoize(obj)
--> 606         self._batch_appends(iter(obj))
    607 
    608     dispatch[ListType] = save_list

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in _batch_appends(self, items)
    637                 write(MARK)
    638                 for x in tmp:
--> 639                     save(x)
    640                 write(APPENDS)
    641             elif n:

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in save(self, obj)
    284         f = self.dispatch.get(t)
    285         if f:
--> 286             f(self, obj) # Call unbound method with explicit self
    287             return
    288 

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in save_tuple(self, obj)
    552         if n <= 3 and proto >= 2:
    553             for element in obj:
--> 554                 save(element)
    555             # Subtle.  Same as in the big comment below.
    556             if id(obj) in memo:

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in save(self, obj)
    284         f = self.dispatch.get(t)
    285         if f:
--> 286             f(self, obj) # Call unbound method with explicit self
    287             return
    288 

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in save_tuple(self, obj)
    566         write(MARK)
    567         for element in obj:
--> 568             save(element)
    569 
    570         if id(obj) in memo:

/home/antgr/anaconda2/lib/python2.7/pickle.pyc in save(self, obj)
    284         f = self.dispatch.get(t)
    285         if f:
--> 286             f(self, obj) # Call unbound method with explicit self
    287             return
    288 

/home/antgr/anaconda2/lib/python2.7/site-packages/cloudpickle/cloudpickle.pyc in save_builtin_function(self, obj)
    328         if obj.__module__ == "__builtin__":
    329             return self.save_global(obj)
--> 330         return self.save_function(obj)
    331     dispatch[types.BuiltinFunctionType] = save_builtin_function
    332 

/home/antgr/anaconda2/lib/python2.7/site-packages/cloudpickle/cloudpickle.pyc in save_function(self, obj, name)
    201         # reference (as is done in default pickler), via save_function_tuple.
    202         if (islambda(obj)
--> 203                 or getattr(obj.__code__, 'co_filename', None) == '<stdin>'
    204                 or themodule is None):
    205             self.save_function_tuple(obj)

AttributeError: 'builtin_function_or_method' object has no attribute '__code__'
```

**Comment by jcrist** [comment]
It looks like the issue is in cloudpickle, dealing with serialization of builtin functions. Two things would help with diagnosing this:

- What version of cloudpickle do you have installed? Just do:
```python
import cloudpickle
cloudpickle.__version__
```
- What is `dsk`? A reproducible example will be helpful.

**Comment by antgr** [comment]
Thanks.
version of cloudpickle:

In [1]: import cloudpickle
In [2]: cloudpickle.__version__
Out[2]: '0.2.1'

I used this tutorial
http://dask.pydata.org/en/latest/optimize.html,
>>> from __future__ import print_function

>>> def print_and_return(string):
        print(string)
        return string

>>> format_str = 'word list has {0} occurrences of {1}, out of {2} words'

>>> dsk = {'words': 'apple orange apple pear orange pear pear',
           'nwords': (len, (str.split, 'words')),
           'val1': 'orange',
           'val2': 'apple',
           'val3': 'pear',
           'count1': (str.count, 'words', 'val1'),
           'count2': (str.count, 'words', 'val2'),
           'count3': (str.count, 'words', 'val3'),
           'out1': (format_str.format, 'count1', 'val1', 'nwords'),
           'out2': (format_str.format, 'count2', 'val2', 'nwords'),
           'out3': (format_str.format, 'count3', 'val3', 'nwords'),
           'print1': (print_and_return, 'out1'),
           'print2': (print_and_return, 'out2'),
           'print3': (print_and_return, 'out3')}

>>> from dask.multiprocessing import get

>>> results = get(dsk, ['print1', 'print2'])    <---- here is the error for me.
word list has 3 occurrences of pear, out of 7 words
word list has 2 occurrences of apple, out of 7 words
word list has 2 occurrences of orange, out of 7 words

etc.

Also I have installed the following packages, in case any info from there is needed.

# packages in environment at /home/antgr/anaconda2:
#
_license                  1.1                      py27_1    defaults
abstract-rendering        0.5.1               np110py27_0    defaults
alabaster                 0.7.7                    py27_0    defaults
anaconda                  4.0.0               np110py27_0    defaults
anaconda-client           1.4.0                    py27_0    defaults
anaconda-navigator        1.1.0                    py27_0    defaults
argcomplete               1.0.0                    py27_1    defaults
astropy                   1.1.2               np110py27_0    defaults
babel                     2.2.0                    py27_0    defaults
backports                 1.0                      py27_0    defaults
backports-abc             0.4                       <pip>
backports.shutil-get-terminal-size 1.0.0                     <pip>
backports.ssl-match-hostname 3.4.0.2                   <pip>
backports_abc             0.4                      py27_0    defaults
beautifulsoup4            4.4.1                    py27_0    defaults
bitarray                  0.8.1                    py27_0    defaults
blaze                     0.9.1                    py27_0    defaults
blaze-core                0.9.0                    py27_0    defaults
bokeh                     0.11.1                   py27_0    defaults
boto                      2.39.0                   py27_0    defaults
bottleneck                1.0.0               np110py27_0    defaults
cairo                     1.12.18                       6    defaults
cdecimal                  2.3                      py27_0    defaults
cffi                      1.5.2                    py27_0    defaults
chest                     0.2.3                    py27_0    defaults
cloudpickle               0.2.1                    py27_0    defaults
clyent                    1.2.1                    py27_0    defaults
colorama                  0.3.7                    py27_0    defaults
conda                     4.0.6                    py27_0    defaults
conda-build               1.20.1                   py27_0    defaults
conda-env                 2.4.5                    py27_0    defaults
conda-manager             0.3.1                    py27_0    defaults
configobj                 5.0.6                    py27_0    defaults
configparser              3.5.0b2                  py27_1    defaults
cryptography              1.3                      py27_0    defaults
curl                      7.45.0                        0    defaults
cycler                    0.10.0                   py27_0    defaults
cython                    0.20.1                   py27_0    defaults
cytoolz                   0.7.5                    py27_0    defaults
dask                      0.9.0                    py27_2    conda-forge
datashape                 0.5.1                    py27_0    defaults
decorator                 4.0.9                    py27_0    defaults
dill                      0.2.4                    py27_0    defaults
docutils                  0.12                     py27_0    defaults
dynd                      0.7.3.dev1                <pip>
dynd-python               0.7.2                    py27_0    defaults
entrypoints               0.2                      py27_1    defaults
enum34                    1.1.2                    py27_0    defaults
et-xmlfile                1.0.1                     <pip>
et_xmlfile                1.0.1                    py27_0    defaults
fastcache                 1.0.2                    py27_0    defaults
flask                     0.10.1                   py27_1    defaults
flask-cors                2.1.2                    py27_0    defaults
fontconfig                2.11.1                        5    defaults
freetype                  2.5.5                         0    defaults
funcsigs                  0.4                      py27_0    defaults
functools32               3.2.3.2                  py27_0    defaults
futures                   3.0.3                    py27_0    defaults
get_terminal_size         1.0.0                    py27_0    defaults
gevent                    1.1.0                    py27_0    defaults
gevent-websocket          0.9.5                    py27_1    defaults
glib                      2.43.0                        0    defaults
graphviz                  2.38.0                        1    defaults
greenlet                  0.4.9                    py27_0    defaults
grin                      1.2.1                    py27_1    defaults
h5py                      2.5.0               np110py27_4    defaults
harfbuzz                  0.9.39                        0    defaults
hdf5                      1.8.15.1                      2    defaults
heapdict                  1.0.0                    py27_0    defaults
idna                      2.0                      py27_0    defaults
imagesize                 0.7.0                    py27_0    defaults
ipaddress                 1.0.14                   py27_0    defaults
ipykernel                 4.3.1                    py27_0    defaults
ipython                   4.1.2                    py27_1    defaults
ipython-genutils          0.1.0                     <pip>
ipython-notebook          4.0.4                    py27_0    defaults
ipython-qtconsole         4.0.1                    py27_0    defaults
ipython_genutils          0.1.0                    py27_0    defaults
ipywidgets                4.1.1                    py27_0    defaults
itsdangerous              0.24                     py27_0    defaults
jbig                      2.1                           0    defaults
jdcal                     1.2                      py27_0    defaults
jedi                      0.9.0                    py27_0    defaults
jinja2                    2.8                      py27_0    defaults
jpeg                      8d                            0    defaults
jsonschema                2.4.0                    py27_0    defaults
jupyter                   1.0.0                    py27_2    defaults
jupyter-client            4.2.2                     <pip>
jupyter-console           4.1.1                     <pip>
jupyter-core              4.1.0                     <pip>
jupyter_client            4.2.2                    py27_0    defaults
jupyter_console           4.1.1                    py27_0    defaults
jupyter_core              4.1.0                    py27_0    defaults
keras                     1.0.0                    py27_0    jaikumarm
kivy                      1.8.0                    py27_0    judowill
kivy-garden               0.1.1                    py27_0    auto
libdynd                   0.7.2                         0    defaults
libffi                    3.0.13                        0    defaults
libgfortran               3.0                           0    defaults
libpng                    1.6.17                        0    defaults
libsodium                 1.0.3                         0    defaults
libtiff                   4.0.6                         1    defaults
libxml2                   2.9.2                         0    defaults
libxslt                   1.1.28                        0    defaults
llvmlite                  0.9.0                    py27_0    defaults
locket                    0.2.0                    py27_0    defaults
lxml                      3.6.0                    py27_0    defaults
mako                      1.0.4                    py27_0    defaults
markupsafe                0.23                     py27_0    defaults
matplotlib                1.5.1               np110py27_0    defaults
mistune                   0.7.2                    py27_0    defaults
mkl                       11.3.1                        0    defaults
mkl-service               1.1.2                    py27_0    defaults
mpmath                    0.19                     py27_0    defaults
multipledispatch          0.4.8                    py27_0    defaults
nbconvert                 4.1.0                    py27_0    defaults
nbformat                  4.0.1                    py27_0    defaults
networkx                  1.11                     py27_0    defaults
nltk                      3.2                      py27_0    defaults
nose                      1.3.7                    py27_0    defaults
notebook                  4.1.0                    py27_1    defaults
numba                     0.24.0              np110py27_0    defaults
numexpr                   2.5                 np110py27_0    defaults
numpy                     1.10.4                   py27_1    defaults
odo                       0.4.2                    py27_0    defaults
openblas                  0.2.14                        4    defaults
openpyxl                  2.3.2                    py27_0    defaults
openssl                   1.0.2d                        0    defaults
pandas                    0.18.0              np110py27_0    defaults
pango                     1.39.0                        0    defaults
partd                     0.3.3                    py27_0    defaults
patchelf                  0.8                           0    defaults
path.py                   8.1.2                    py27_1    defaults
patsy                     0.4.0               np110py27_0    defaults
pep8                      1.7.0                    py27_0    defaults
pexpect                   4.0.1                    py27_0    defaults
pickleshare               0.5                      py27_0    defaults
pillow                    3.1.1                    py27_0    defaults
pip                       8.1.1                    py27_1    defaults
pixman                    0.32.6                        0    defaults
ply                       3.8                      py27_0    defaults
protobuf                  3.0.0b2                  py27_1    jjhelmus
psutil                    4.1.0                    py27_0    defaults
ptyprocess                0.5                      py27_0    defaults
py                        1.4.31                   py27_0    defaults
py2cairo                  1.10.0                   py27_2    defaults
pyasn1                    0.1.9                    py27_0    defaults
pycairo                   1.10.0                   py27_0    defaults
pycosat                   0.6.1                    py27_0    defaults
pycparser                 2.14                     py27_0    defaults
pycrypto                  2.6.1                    py27_0    defaults
pycurl                    7.19.5.3                 py27_0    defaults
pydot                     1.0.28                   py27_0    anaconda
pyflakes                  1.1.0                    py27_0    defaults
pygame                    1.9.2                    py27_1    tlatorre
pygments                  2.1.1                    py27_0    defaults
pygpu                     2016.4.1                 py27_1    trung
pymc                      2.3.6               np110py27_1    defaults
pyopenssl                 0.15.1                   py27_2    defaults
pyparsing                 1.5.6                    py27_0    anaconda
pyqt                      4.11.4                   py27_1    defaults
pytables                  3.2.2               np110py27_1    defaults
pytest                    2.8.5                    py27_0    defaults
python                    2.7.11                        0    defaults
python-dateutil           2.5.1                    py27_0    defaults
python-wget               3.2                      py27_0    bioconda
pytz                      2016.2                   py27_0    defaults
pyyaml                    3.11                     py27_1    defaults
pyzmq                     15.2.0                   py27_0    defaults
qt                        4.8.7                         1    defaults
qtawesome                 0.3.2                    py27_0    defaults
qtconsole                 4.2.0                    py27_0    defaults
qtpy                      1.0                      py27_0    defaults
readline                  6.2                           2    defaults
redis                     2.6.9                         0    defaults
redis-py                  2.10.3                   py27_0    defaults
requests                  2.9.1                    py27_0    defaults
rope                      0.9.4                    py27_1    defaults
scikit-image              0.12.3              np110py27_0    defaults
scikit-learn              0.17.1              np110py27_0    defaults
scipy                     0.17.0              np110py27_2    defaults
sdl                       2.0.1                         1    tlatorre
seaborn                   0.7.0                    py27_0    defaults
setuptools                20.3                     py27_0    defaults
simplegeneric             0.8.1                    py27_0    defaults
singledispatch            3.4.0.3                  py27_0    defaults
sip                       4.16.9                   py27_0    defaults
six                       1.10.0                   py27_0    defaults
snowballstemmer           1.2.1                    py27_0    defaults
sockjs-tornado            1.0.1                    py27_0    defaults
sphinx                    1.3.5                    py27_0    defaults
sphinx-rtd-theme          0.1.9                     <pip>
sphinx_rtd_theme          0.1.9                    py27_0    defaults
spyder                    2.3.8                    py27_1    defaults
spyder-app                2.3.8                    py27_0    defaults
sqlalchemy                1.0.12                   py27_0    defaults
sqlite                    3.9.2                         0    defaults
ssl_match_hostname        3.4.0.2                  py27_0    defaults
statsmodels               0.6.1               np110py27_0    defaults
sympy                     1.0                      py27_0    defaults
tables                    3.2.2                     <pip>
tensorflow                0.8.0rc0                 py27_0    jjhelmus
terminado                 0.5                      py27_1    defaults
theano                    0.8.2                    py27_0    trung
tk                        8.5.18                        0    defaults
toolz                     0.7.4                    py27_0    defaults
tornado                   4.3                      py27_0    defaults
traitlets                 4.2.1                    py27_0    defaults
ujson                     1.35                     py27_0    defaults
unicodecsv                0.14.1                   py27_0    defaults
util-linux                2.21                          0    defaults
werkzeug                  0.11.4                   py27_0    defaults
wget                      3.2                       <pip>
wheel                     0.29.0                   py27_0    defaults
xlrd                      0.9.4                    py27_0    defaults
xlsxwriter                0.8.4                    py27_0    defaults
xlwt                      1.0.0                    py27_0    defaults
xz                        5.0.5                         1    defaults
yaml                      0.1.6                         0    defaults
zeromq                    4.1.3                         0    defaults
zlib                      1.2.8                         0    defaults

**Comment by mrocklin** [comment]
Raised issue here: https://github.com/cloudpipe/cloudpickle/issues/61

I'll take a look probably in the next few days

**Comment by jcrist** [comment]
The issue was a serialization one with cloudpickle, as well as our docs being a bit outdated. Docs are fixed in #1173.
